L1Trigger/Phase2L1ParticleFlow/interface/COEFile.h

```diff
@@ -0,0 +1,43 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_CoeFile_h
+#define L1Trigger_Phase2L1ParticleFlow_CoeFile_h
+
+// system include files
+#include <vector>
+#include <string>
+#include <numeric>
+#include <boost/dynamic_bitset.hpp>
+#include <boost/multiprecision/cpp_int.hpp>
+
+// user include files
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/DiscretePFInputs.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/Region.h"
+
+namespace l1tpf_impl {
+  class COEFile {
+  public:
+    COEFile(const edm::ParameterSet&);
+    ~COEFile();
+
+    void close() { fclose(file); }
+    template <typename T>
+    bool getBit(T value, unsigned bit) {
+      return (value >> bit) & 1;
+    }
+    bool is_open() { return (file != nullptr); }
+    void writeHeaderToFile();
+    void writeTracksToFile(const std::vector<Region>& regions, bool print = false);
+
+  protected:
+    FILE* file;
```

Comment:
plain FILE objects have been used for some of these dumpers because the dumps have to be read from other software where plain C is easier/safer.
I'll add the include

Reply:
I just want to point out `fstream`-written files can be read with plain C.